managers/docker-manager/DockerService.js

```diff
@@ -0,0 +1,801 @@
+'use strict';
+
+const _ = require('lodash');
+const Promise = require('bluebird');
+const logger = require('../../common/logger');
+const errors = require('../../common/errors');
+const utils = require('../../common/utils');
+const docker = require('../../data-access-layer/docker');
+const catalog = require('../../common/models').catalog;
+const Timeout = errors.Timeout;
+const ContainerStartError = errors.ContainerStartError;
+const ServiceInstanceAlreadyExists = errors.ServiceInstanceAlreadyExists;
+const ServiceInstanceNotFound = errors.ServiceInstanceNotFound;
+const CONST = require('../../common/constants');
+const assert = require('assert');
+const config = require('../../common/config');
+const BaseService = require('../BaseService');
+const BasePlatformManager = require('../../broker/lib/fabrik/BasePlatformManager');
+const DockerImageLoaderService = require('./DockerImageLoaderService');
+
+const DockerError = {
+  NotFound: {
+    statusCode: CONST.HTTP_STATUS_CODE.NOT_FOUND
+  },
+  Conflict: {
+    statusCode: CONST.HTTP_STATUS_CODE.CONFLICT
+  },
+  ServerError: {
+    statusCode: CONST.HTTP_STATUS_CODE.INTERNAL_SERVER_ERROR
+  }
+};
+
+class DockerService extends BaseService {
+  constructor(guid, plan) {
+    super(plan);
+    this.guid = guid;
+    this.plan = plan;
+    this.platformManager = undefined;
+    this.prefix = CONST.SERVICE_FABRIK_PREFIX;
+    this.credentials = docker.createCredentials(this.plan.credentials);
+    this.imageInfo = undefined;
+    this.docker = docker.createClient();
+    this.container = this.docker.getContainer(this.containerName);
+    this.containerInfo = undefined;
+  }
+
+  assignPlatformManager(platformManager) {
```

Comment:
should go!

Reply:
Platform managers would still remain. SF2.0 managers are for the provisioning use case. Platform managers are for the consuming platform specific use case like SG creation. Cant be removed.